### ISSUE TITLE
GPU kernel for SVD

### ISSUE THREAD

**Issue by rana-alshaikh** [tensorflow/tensorflow#11588]
### System information
-Linux Ubuntu 16.04
- TensorFlow installed from (source )
- python v3.5 
- **GPU 8gb**:

### Describe the problem
I tried to run code that used tf.svd on GPU but it gives me an error. Is it correct that tensorflow does not have GPU kernels for SVD?

### Source code / logs
[[Node: Svd = Svd[T=DT_FLOAT, compute_uv=false, full_matrices=false](x)]]
[[Node: map_1/while/nuclear_norm_09660f0e_1 = nuclear_norm_09660f0e[_device="/job:localhost/replica:0/task:0/gpu:0"](map_1/while/Reshape_1, ^map_1/while/Identity)]]
[[Node: Adagrad_1/update/_64 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/gpu:0", send_device_incarnation=1, tensor_name="edge_690_Adagrad_1/update", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"]()]]

**Comment by asimshankar** [comment]
Nope, currently there is no GPU kernel registered for SVD. 

I don't believe anyone is actively working on this, in which case contributions will be welcome.
@rmlarsen might want to chime in with more detail.

**Comment by rana-alshaikh** [comment]
Ok
is it possible to run  part of the loss function on CPU (the SVD part) and the other parts on GPU
I tried with tf.device it still gives me an error

**Comment by gautam1858** [comment]
It would be tough as there are no such pipelines created

**Comment by asimshankar** [comment]
@rana-alshaikh : Not entirely sure what you mean, but yes, it is possible to place individual operations on specific devices. For details see https://www.tensorflow.org/tutorials/using_gpu#manual_device_placement

If you're running into trouble, I'd suggest posting on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow).

**Comment by gautam1858** [comment]
@asimshankar What he is telling is he wants to run some only loss funciton on CPU and later run other training things on GPU which I am telling is not possible, you are telling usage of multiple GPU he is asking multiple CPU an GPU at the same time

**Comment by asimshankar** [comment]
I might be misunderstanding, but why can't multiple devices be mixed? For example:

```python

import tensorflow as tf

with tf.device('/gpu:0'):
  # All these operations will run on GPU
  x = tf.constant(1.0, name='X')
  y = tf.add(x, x, name='Y')
with tf.device('/cpu:0'):
  # This will copy the result of `y` to CPU and execute this multiple on CPU
  z = tf.multiply(y, 2, name='Z')

# Log device placement to see where ops are run
with tf.Session(config=tf.ConfigProto(log_device_placement=True)) as sess:
  print(sess.run(z))
```

Will result in the following:

```
Y: (Add): /job:localhost/replica:0/task:0/gpu:0
Z: (Mul): /job:localhost/replica:0/task:0/cpu:0
X: (Const): /job:localhost/replica:0/task:0/gpu:0
4.0
```

**Comment by gautam1858** [comment]
@asimshankar Wow cool, sorry I was not aware  :) And can I do this in parallel the same thing that you did above instead of sequential for two different functions ? For example in GANs, I will run in parallel i.e the generator loss in CPU and discriminator loss in GPU. Can this be done as well ?

**Comment by mudream4869** [comment]
GPU for SVD seems done now? https://github.com/tensorflow/tensorflow/issues/13603

**Comment by wt-huang** [comment]
Closing this since the feature is added
